### PR TITLE
Unit TC / Multi-Stream Mux with Different Rates

### PR DESCRIPTION
**Pull request by myungjoo**
This tests:
```
SRC1 10FPS --> t_converter --> t_filter:X (1) --+
SRC2 25FPS --> t_converter --> t_filter:X (2) --+--> t_mux (sync=slowest) --> t_filter:X (3) --> FILESINK
```
And see if X (3) is processing at the rate of 10FPS.

This is partial adress of 1st item of #739 

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

> [Tests] Add multi-stream sync TC for #739
    
    Using the "nnscustom_framecounter.so", check if the
    1st item of #739 is working properly if the sync policy
    is "slowest".
  
  
> [Tests] Add a custom filter for testcase of #739
    
    For testcases in #739, we need a custom filter
    that shows frame numbers.

> Clean up customfilter dependencies, support non-single-binary mode
    
    Building custom filter requires "libtensor_filter.so" only.
    Other libraries used by nnstreamer are not necessary for custom filters.
    Thus, let them depend on libtensor_filter (or libnnstreamer.so) only.
    
    Plus, by defining "CUSTOMFILTERDEP" CMake variable,
    support both SINGLEBINARY and NON-SINGLEBINARY.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,10 @@ ADD_EXECUTABLE(unittest_sink nnstreamer_sink/unittest_sink.cpp ${gtestSrc})
 TARGET_INCLUDE_DIRECTORIES(unittest_sink PUBLIC ${pkgs_INCLUDE_DIRS} ${testpkgs_INCLUDE_DIRS} ${gtestInc})
 TARGET_LINK_LIBRARIES(unittest_sink ${pkgs_LIBRARIES} ${testpkgs_LIBRARIES} ${gtestLink})
 
+# Unit test uses this custom filter
+ADD_LIBRARY(nnscustom_framecounter SHARED nnstreamer_sink/nnscustom_framecounter.c)
+TARGET_LINK_LIBRARIES(nnscustom_framecounter nnstreamer)
```
Comment (jaeyun-jung) [inline tests/CMakeLists.txt]: nnstreamer defined only SINGLE_BINARY=ON (use ${pkgs_LIBRARIES} instead?)

Comment (myungjoo) [reply tests/CMakeLists.txt]: Anyway, pkg_LIBRARIES won't be appropraite for custom filters. They only need ```libtensor_filter.so```.
Plus, with SINGLE_BINARY=OFF, other custom filters are failing to build :)
I'll fix it with another commit

Comment (myungjoo) [reply tests/CMakeLists.txt]: Refer to the third commit in this PR.

File: tests/nnstreamer_sink/nnscustom_framecounter.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    out_info->info[0].dimension[i] = 1;
+  out_info->info[0].type = _NNS_UINT32;
+
+  data->copy = 1;
+  for (i = 0; i < NNS_TENSOR_RANK_LIMIT; i++) {
+    if (in_info->info[0].dimension[i] != 1) {
+      data->copy = 0;
+      break;
+    }
+    if (in_info->info[0].type != _NNS_UINT32)
```
Comment (jaeyun-jung) [inline tests/nnstreamer_sink/nnscustom_framecounter.c]: better to move this condition outside the loop

Comment (myungjoo) [reply tests/nnstreamer_sink/nnscustom_framecounter.c]: Haha. yes :)

**Pull request merged**